packages/gatsby/src/schema/types/derived-types.ts

```diff
@@ -29,9 +29,37 @@
  *   user explicitly defines `FooFields` type (via `createTypes` call) it won't be considered
  *   a derived type
  */
-const { ObjectTypeComposer, InterfaceTypeComposer } = require(`graphql-compose`)
+import {
+  ObjectTypeComposer,
+  InterfaceTypeComposer,
+  ScalarTypeComposer,
+  SchemaComposer,
+  InputTypeComposer,
+  EnumTypeComposer,
+  UnionTypeComposer,
+} from "graphql-compose"
 
-const clearDerivedTypes = ({ schemaComposer, typeComposer }) => {
+type AllTypeComposer =
+  | ObjectTypeComposer
+  | InputTypeComposer
+  | EnumTypeComposer
+  | InterfaceTypeComposer
+  | UnionTypeComposer
+  | ScalarTypeComposer
+
+const getDerivedTypes = ({
+  typeComposer,
+}: {
+  typeComposer: AllTypeComposer
+}): any => typeComposer.getExtension(`derivedTypes`) || new Set()
```

Comment:
I'm looking into the return type here.

I can see in https://graphql-compose.github.io/docs/api/ObjectTypeComposer.html that the returned type is `Extensions`. Unfortunately that does not describe the type. The default being a `Set` is a hint, but a `Set` of what ...

https://github.com/graphql-compose/graphql-compose/blob/master/src/utils/definitions.d.ts#L16 implies that it is:
```ts
export type Extensions = {
  [key: string]: any;
  directives?: ExtensionsDirective[];
};
```
But that doesn't seem right since we default to a `Set`, AND we call `.add` on it in `addDerivedType()` (see later in this file). 

```js
  const derivedTypes = getDerivedTypes({ typeComposer })
  typeComposer.setExtension(`derivedTypes`, derivedTypes.add(derivedTypeName))
```

So this feels a little troubling, to say the least.

@vladar do you perhaps have an insight here? Do we have a bug in our code or am I looking at the wrong docs?

Reply:
> I can see in https://graphql-compose.github.io/docs/api/ObjectTypeComposer.html that the returned type is `Extensions`

It is the returned type of `getExtensions` method. But we are looking at `getExtension` which has `any` as the returned type. Extension can be basically anything we want. In our case `derivedTypes` extension is a `Set | void`.

Reply:
I tried to fix it.